File: src/python/src/main/marmara/dto/StudentGetDTo.py

from datetime import datetime
from uuid import UUID

class StudentGetDTO:
    def __init__(self, uuid=None, name=None, surname=None, email=None, birth_date=None, student_id=None, year_enrolled=None, advisor=None, semester=None):
        self.uuid = uuid
        self.name = name
        self.surname = surname
        self.email = email
        self.birth_date = birth_date
        self.student_id = student_id
        self.year_enrolled = year_enrolled
        self.advisor = advisor
        self.semester = semester

    def __str__(self):
        return (f"StudentGetDTO{{uuid={self.uuid}, name='{self.name}', surname='{self.surname}', "
                f"email='{self.email}', birth_date={self.birth_date}, student_id={self.student_id}, "
                f"year_enrolled={self.year_enrolled}, advisor='{self.advisor}', semester={self.semester}}}")

    @property
    def uuid(self):
        return self._uuid

    @uuid.setter
    def uuid(self, value):
        if isinstance(value, UUID):
            self._uuid = value
        elif isinstance(value, str):
            self._uuid = UUID(value)
        else:
            self._uuid = value

    @property
    def birth_date(self):
        return self._birth_date

    @birth_date.setter
    def birth_date(self, value):
        if isinstance(value, datetime):
            self._birth_date = value
        elif isinstance(value, str):
            self._birth_date = datetime.fromisoformat(value)
        else:
            self._birth_date = value

File: java/src/main/java/edu/marmara/dto/StudentGetDTO.java
package edu.marmara.dto;


import java.util.Date;
import java.util.UUID;

public class StudentGetDTO {
    private UUID uuid;
    private String name;
    private String surname;
    private String email;
    private Date birthDate;
    private Long studentId;
    private Integer yearEnrolled;
    private String advisor;
    private Integer semester;

    public StudentGetDTO() {
    }

    public StudentGetDTO(UUID uuid, String name, String surname, String email, Date birthDate, Long studentId, Integer yearEnrolled, String advisor, Integer semester) {
        this.uuid = uuid;
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.birthDate = birthDate;
        this.studentId = studentId;
        this.yearEnrolled = yearEnrolled;
        this.advisor = advisor;
        this.semester = semester;
    }

    @Override
    public String toString() {
        return "StudentGetDTO{" +
                "uuid=" + uuid +
                ", name='" + name + '\'' +
                ", surname='" + surname + '\'' +
                ", email='" + email + '\'' +
                ", birthDate=" + birthDate +
                ", studentId=" + studentId +
                ", yearEnrolled=" + yearEnrolled +
                ", advisor='" + advisor + '\'' +
                ", semester=" + semester +
                '}';
    }

    public UUID getUuid() {

        return uuid;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Date getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public Long getStudentId() {
        return studentId;
    }

    public void setStudentId(Long studentId) {
        this.studentId = studentId;
    }

    public Integer getYearEnrolled() {
        return yearEnrolled;
    }

    public void setYearEnrolled(Integer yearEnrolled) {
        this.yearEnrolled = yearEnrolled;
    }

    public String getAdvisor() {
        return advisor;
    }

    public void setAdvisor(String advisor) {
        this.advisor = advisor;
    }

    public Integer getSemester() {
        return semester;
    }

    public void setSemester(Integer semester) {
        this.semester = semester;
    }
}
